packages/hardhat-verify/src/internal/sourcify.ts

```diff
@@ -14,8 +14,17 @@ import { ValidationResponse } from "./utilities";
 export class Sourcify {
   public apiUrl: string = "https://sourcify.dev/server";
   public browserUrl: string = "https://repo.sourcify.dev";
+  public chainId: number;
 
-  constructor(public chainId: number) {}
+  constructor(chainId: number, apiUrl?: string, browserUrl?: string) {
+    this.chainId = chainId;
+    if (apiUrl !== undefined) {
+      this.apiUrl = apiUrl;
+    }
+    if (browserUrl !== undefined) {
+      this.browserUrl = browserUrl;
+    }
+  }
```

Comment:
As `apiUrl` and `browserUrl` are passed as parameters to the constructor and have default values in the config, the initialization of the variables in lines 15 and 16, and the checks in 21 and 24 are not needed. And with these changes, the parameters in the constructor should be required, not optional. The entire section could be just:

```

export class Sourcify {
  constructor(
    public chainId: number,
    public apiUrl: string,
    public browserUrl: string
  ) {}
```

Reply:
In the types I set them optional parameters

```
export interface SourcifyConfig {
  enabled: boolean;
  apiUrl?: string;
  browserUrl?: string;
}
```

Making them mandatory will force the user to set them in the hardhat config?

Reply:
That is fine. You are going to make them required for the `constructor`, but you'll leave them optional in the types. If the user doesn't set them, we use the default value specified in the `defaultSourcifyConfig`:
```
config.sourcify = { ...defaultSourcifyConfig, ...userSourcifyConfig };
```

Reply:
ok, done. I passed them using the non-null assertion operator, if the user will pass an empty string a network error will be presented.